build.gradle.kts

```diff
@@ -0,0 +1,132 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+import java.io.ByteArrayOutputStream
+import java.util.Date
+
+plugins {
+    java
+    `maven-publish`
+    checkstyle
+
+    id("com.github.johnrengelman.shadow") version "7.0.0"
+}
+
+val buildExtras = System.getenv("BUILD_EXTRAS")?.toBoolean() ?: false
+val apiVersion = "1.16.5-R0.1-SNAPSHOT"
+
+repositories {
+    mavenLocal()
+    mavenCentral()
+
+    maven("https://repo.glowstone.net/repository/maven-public/")
+    maven("https://repo.glowstone.net/repository/snapshots/")
+    maven("https://libraries.minecraft.net")
+}
+
+dependencies {
```

Comment:
This won't do - use [version catalogs](https://docs.gradle.org/current/userguide/platforms.html#sub:version-catalog).

We make use of them at Kord Extensions, and it's much, much easier to manage deps this way. For examples:

* [settings.gradle.kts](https://github.com/Kord-Extensions/kord-extensions/blob/root/settings.gradle.kts)
* [libs.version.toml](https://github.com/Kord-Extensions/kord-extensions/blob/root/libs.versions.toml)
* [build.gradle.kts](https://github.com/Kord-Extensions/kord-extensions/blob/root/kord-extensions/build.gradle.kts)

Reply:
I have a question regarding this. How is the support of GitHub dependabot on this? I would prefer putting versions in settings.gradle.kts and let dependabot update them. What do you think?

Reply:
Gradle support in dependabot is already fairly lacking, although I do wonder if it's that much of a deal in our case - I don't find myself getting many emails for it here, if any

Reply:
I'm doing it! :P